modin/utils.py

```diff
@@ -588,8 +588,8 @@ def try_cast_to_pandas(obj: Any, squeeze: bool = False) -> Any:
     object
         Converted object.
     """
-    if isinstance(obj, SupportsPublicToPandas):
-        result = obj.to_pandas()
+    if isinstance(obj, SupportsPublicToPandas) or hasattr(obj, "modin"):
+        result = obj.modin.to_pandas() if hasattr(obj, "modin") else obj.to_pandas()
```

Comment:
`obj.to_pandas()`. Do high level object still have a public method to_pandas?

Reply:
ping

Reply:
Looks like we don't. Let's see how things are going in CI.

Reply:
If those do not have the public method to_pandas, else branch is likely going to be for lower level objects such as QCs.

Reply:
> Do high level objects still have a public method to_pandas?

Query compilers still have `to_pandas` and use `try_cast_to_pandas` function.

Reply:
The next detail worries me:
```python
>>> from modin.utils import SupportsPublicToPandas 
>>> isinstance(pd.Series([1,2,3,4]), SupportsPublicToPandas) 
UserWarning: Distributing <class 'list'> object. This may take some time.
False
>>> isinstance(pd.Series([1,2,3,4]).modin, SupportsPublicToPandas) 
UserWarning: Distributing <class 'list'> object. This may take some time.
True
```

Reply:
I thought that some users could use this class to check whether an object can be converted to a pandas. Then I remembered that there was no release yet, so everything is fine.